### PR TITLE
fix: Use lock and map instead of concurrentMap

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #31209

### CODE REVIEW

File: internal/datanode/event_manager.go [hunk 1]
```diff
@@ -402,3 +396,48 @@ func GetLiteChannelWatchInfo(watchInfo *datapb.ChannelWatchInfo) *datapb.Channel
 		Progress:  watchInfo.GetProgress(),
 	}
 }
+
+type EventManager struct {
+	channelGurad    sync.Mutex
```
Comment (congqixia) [inline internal/datanode/event_manager.go:401]: NIT: typo
```suggestion
	channelGuard    sync.Mutex
```

File: internal/datanode/event_manager.go [hunk 2]
```diff
@@ -402,3 +396,48 @@ func GetLiteChannelWatchInfo(watchInfo *datapb.ChannelWatchInfo) *datapb.Channel
...
+
+func (m *EventManager) Remove(channel string) {
+	m.channelGurad.Lock()
+	defer m.channelGurad.Unlock()
+
+	if eManager, got := m.channelManagers[channel]; got {
+		eManager.Close()
+	}
+
+	delete(m.channelManagers, channel)
```
Comment (congqixia) [inline internal/datanode/event_manager.go:433]:
```suggestion
	m.channelGurad.Lock()
	eManager, got := m.channelManagers[channel]
	delete(m.channelManagers, channel)
	m.channelGurad.Unlock()

	if got {
		eManager.Close()
	}
```

File: internal/datanode/event_manager.go [hunk 3]
```diff
@@ -402,3 +396,48 @@ func GetLiteChannelWatchInfo(watchInfo *datapb.ChannelWatchInfo) *datapb.Channel
...
+}
+
+func (m *EventManager) CloseAll() {
+	m.channelGurad.Lock()
+	defer m.channelGurad.Unlock()
+
+	for _, eManager := range m.channelManagers {
+		eManager.Close()
+	}
```
Comment (congqixia) [inline internal/datanode/event_manager.go:442]: remove manager and close

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by XuanYang-cn** [comment]
failed at LoadWithMmap

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**